sdk/download.go

```diff
@@ -151,3 +153,128 @@ func DownloadSDK(version string) error {
 	tarURL := TarballURL(version)
 	return DownloadSignedFile(tarFile, tarURL)
 }
+
+// false if both files do not exist
+func cmpFileBytes(file1, file2 string) (bool, error) {
+	info1, err := os.Stat(file1)
+	if err != nil {
+		return false, err
+	}
+	info2, err := os.Stat(file2)
+	if err != nil {
+		return false, err
+	}
+	if info1.Size() != info2.Size() {
+		return false, nil
+	}
+
+	f1, err := os.Open(file1)
+	if err != nil {
+		return false, err
+	}
+	defer f1.Close()
+	f2, err := os.Open(file2)
+	if err != nil {
+		return false, err
+	}
+	defer f2.Close()
+
+	const defaultBufSize = 4096 // same as bufio
+	buf1 := make([]byte, defaultBufSize)
+	buf2 := make([]byte, defaultBufSize)
+
+	for {
+		n1, err1 := io.ReadFull(f1, buf1)
+		n2, err2 := io.ReadFull(f2, buf2)
+
+		if err1 == io.EOF && err2 == io.EOF {
+			return true, nil
+		} else if err1 == io.EOF || err2 == io.EOF {
+			return false, nil
+		}
+
+		if err1 == io.ErrUnexpectedEOF && err2 == io.ErrUnexpectedEOF {
+			if n1 == n2 {
+				return bytes.Equal(buf1, buf2), nil
```

Comment:
In the case in which n1 == n2 either it returned early on the first chunk and so the rest of the buffer is zero (and equal) or it successfully compared the last chunk and so, also, the rest of the buffer is equal. If I didn't check n1== n2 and checked the full buffer then this would create a weird and rare failure case.

Your way is probably more readable though so I'll change that.